AssetRipper.Library/Exporters/Audio/AudioMixerExportCollection.cs

```diff
@@ -0,0 +1,272 @@
+					}
+				}
+
+				for (int j = 0; j < snapshotConstant.TransitionIndices.Length; j++)
+				{
+					uint paramIndex = snapshotConstant.TransitionIndices[j];
+					int transitionType = (int)snapshotConstant.TransitionTypes[j];
+					if (indexToGUID.TryGetValue(paramIndex, out var paramGUID))
+					{
+						snapshot.TransitionOverrides_C245[paramGUID] = transitionType;
+					}
+					else
+					{
+						Logger.Warning(LogCategory.Export, $"Snapshot({snapshot.Name_C245}) transition #{paramIndex} has no binding parameter");
+					}
+				}
+			}
+			
+			// generate exposed parameters
+
+			for (int i = 0; i < constants.ExposedParameterIndices.Length; i++)
+			{
+				uint paramIndex = constants.ExposedParameterIndices[i];
+				uint paramNameCrc = constants.ExposedParameterNames[i];
+				if (indexToGUID.TryGetValue(paramIndex, out var paramGUID))
+				{
+					var exposedParam = mixer.ExposedParameters_C241.AddNew();
+					exposedParam.Guid.CopyValues(paramGUID);
+					exposedParam.NameString = $"{CrcUtils.ReverseDigestAscii(paramNameCrc)}";
+				}
+				else
+				{
+					Logger.Warning(LogCategory.Export, $"Exposed parameter #{paramIndex} has no binding parameter");
+				}
+			}
+			
+			// complete mixer controller
+			
+			var groupView = mixer.AudioMixerGroupViews_C241.AddNew();
+			groupView.NameString = "View";
+			foreach (var group in groups)
+			{
+				groupView.Guids.Add(group.GroupID_C243);
+			}
+			mixer.CurrentViewIndex_C241 = 0;
+			mixer.TargetSnapshot_C241.CopyValues(mixer.StartSnapshot_C241);
+		}
+
+		private GUID IndexingNewGUID(uint index, Dictionary<uint, GUID> table)
+		{
+			var guid = (GUID)UnityGUID.NewGuid();
+			if (!table.TryAdd(index, guid))
+			{
+				Logger.Warning(LogCategory.Export, $"Constant index #{index} conflicts with another one.");
+			}
+			return guid;
+		}
+
+		private List<Utf8String> ParseNameBuffer(byte[] buffer)
+		{
+			var names = new List<Utf8String>();
+			int offset = 0;
+			while (buffer[offset] != 0)
+			{
+				int start = offset;
+				while (buffer[++offset] != 0) { }
+
+				var utf8Data = buffer.SubArray(start, offset - start);
+				names.Add(new Utf8String { Data = utf8Data });
+
+				offset++;
+			}
+
+			return names;
+		}
+
+		private uint m_nextExportID;
+		
+		protected override long GenerateExportID(IUnityObjectBase asset)
+		{
+			long exportID = ExportIdHandler.GetMainExportID(asset, m_nextExportID);
+			m_nextExportID += 2;
+			return exportID;
+		}
+		
+		protected override string GetExportExtension(IUnityObjectBase asset) => "mixer";
+		
+		private bool HasAnyEffectParameterNameToRecover { get; set; }
+		
+		protected override bool ExportInner(IProjectAssetContainer container, string filePath, string dirPath)
+		{
+			if (HasAnyEffectParameterNameToRecover)
+			{
+				// Put an editor script file in the exported project. See comments in the file for more detail.
+				var postprocessorScriptPath = System.IO.Path.Combine(dirPath,
```

Comment:
Like I said in the other pr, `var` isn't allowed in my code. I might raise the editor config setting to error in order to enforce this.